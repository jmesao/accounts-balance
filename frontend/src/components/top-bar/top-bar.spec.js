import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
// TODO.- Mock vue-socket.io module and add more tests
import VueSocketIO from 'vue-socket.io';

import TopBar from './top-bar.vue';

describe('TopBar', () => {
  function mountTopBar(props, mutations = {}) {
    const localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(new VueSocketIO({
      debug: true,
      connection: 'http://localhost',
    }));

    const store = new Vuex.Store({
      state: {
        rate: 0,
      },
      mutations: {
        setRate: jest.fn(),
      },
    });

    return shallowMount(TopBar, {
      localVue,
      propsData: {
        ...props,
      },
      store,
    });
  }

  it('should render properly without rate', () => {
    const wrapper = mountTopBar({});
    expect(wrapper.element).toMatchSnapshot();
  });
});
