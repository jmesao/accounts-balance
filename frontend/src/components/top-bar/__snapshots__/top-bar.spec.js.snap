// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should render properly with rate 1`] = `
<div
  class="top-bar"
  rate="5000"
>
  <div
    class="top-bar__menu"
  >
    <span
      class="top-bar__menu__icon material-icons"
    >
      menu
    </span>
  </div>
   
  <div
    class="top-bar__balance"
  >
    <span
      class="top-bar__balance__symbol"
    >
      ฿
    </span>
     : $ -
  
  </div>
   
  <div
    class="top-bar__notification"
  >
    <span
      class="material-icons top-bar__notification__icon"
    >
      notifications_none
    </span>
  </div>
</div>
`;

exports[`TopBar should render properly without rate 1`] = `
<div
  class="top-bar"
>
  <div
    class="top-bar__menu"
  >
    <span
      class="top-bar__menu__icon material-icons"
    >
      menu
    </span>
  </div>
   
  <div
    class="top-bar__balance"
  >
    <span
      class="top-bar__balance__symbol"
    >
      ฿
    </span>
     : $ -
  
  </div>
   
  <div
    class="top-bar__notification"
  >
    <span
      class="material-icons top-bar__notification__icon"
    >
      notifications_none
    </span>
  </div>
</div>
`;
